src/Common/Uuid/UuidRegistry.php

```diff
@@ -305,4 +341,177 @@ public static function isEmptyBinaryUUID($uuidString)
+     * by the limit.
+     * @param int $limit
+     * @return array
+     */
+    public function getUnusedUuidBatch($limit = 10)
+    {
+        if ($limit <= 0)
+        {
+            return [];
+        }
+        $uuids = $this->getUUIDBatch($limit);
+        $dbUUIDs = [];
+
+        if (!$this->disable_tracker) {
+            $sqlColumns = array_map(function($u) { return '`uuid` = ?';}, $uuids);
+            $sqlWhere = implode(" OR ", $sqlColumns);
+            $dbUUIDs = QueryUtils::fetchRecordsNoLog("SELECT `uuid` FROM `uuid_registry` WHERE " . $sqlWhere, $uuids);
+        }
+        if (empty($dbUUIDs)) {
+            if (!empty($this->table_name)) {
+                $sqlColumns = array_map(function($u) { return '`uuid` = ?';}, $uuids);
+                $sqlWhere = implode(" OR ", $sqlColumns);
+                // If using $this->table_name, then ensure uuid is unique in that table
+                $dbUUIDs =  QueryUtils::fetchRecordsNoLog("SELECT `uuid` FROM `" . $this->table_name . "` WHERE " . $sqlWhere, $uuids);
+            } elseif ($this->document_drive === 1) {
+                $sqlColumns = array_map(function($u) { return '`uuid` = ?';}, $uuids);
+                $sqlWhere = implode(" OR ", $sqlColumns);
+                // If using for document labeling on drive, then ensure drive_uuid is unique in documents table
+                $dbUUIDs = QueryUtils::fetchRecordsNoLog("SELECT `drive_uuid` FROM `documents` WHERE " . $sqlWhere, $uuids);
+            }
+        }
+
+        $count = count($dbUUIDs);
+
+        if ($count <= 0) {
+            return $uuids;
+        }
+        $newGenLimit = $limit;
+        if ($count < $limit) {
+            $newGenLimit = $limit - $count;
+        }
+        // generate some new uuids since we had duplicates... which should never happen... but we have this here in
+        // case we do
+        $outstanding = $this->getUnusedUuidBatch($newGenLimit);
+        return array_merge($dbUUIDs, $outstanding);
+    }
+
+    private function createMissingUuidsForTableWithId()
+    {
+        $counter = 0;
+        $count = $this->getTableCountWithMissingUuids();
+        if ($count > 0) {
+            // loop through in batches of 1000
+            // generate min(1000, $count)
+            // generate bulk insert statement
+            $gen_count = min($count, self::UUID_MAX_BATCH_COUNT);
+            $batchUUids = $this->getUnusedUuidBatch($gen_count);
+            $ids = QueryUtils::fetchRecords("SELECT " . $this->table_id . " FROM `" . $this->table_name . "` WHERE `uuid` IS NULL OR `uuid` = '' OR `uuid` = '\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0' LIMIT " . $gen_count);
+            $this->insertUuidsIntoRegistry($batchUUids);
+            for ($i = 0; $i < $gen_count; $i++) {
+                // do single updates
+                sqlQueryNoLog("UPDATE `" . $this->table_name . "` SET `uuid` = ? WHERE `" . $this->table_id . "` = ?", [$batchUUids[$i], $ids[$i][$this->table_id]]);
```

Comment:
seems odd to be using sqlQueryNoLog function here since that is meant to return a single row as array (so does an Fetchrow process on it). Maybe sqlStatementNoLog instead (possibly save a nano of a nano second :) )?

Reply:
This line was copied from the original createMissingUuids. I can change it to sqlStatementNoLog

Reply:
Lets change it and win back a nanosecond or 2 :)